service/dap/server.go

```diff
@@ -1233,6 +1258,7 @@ func (s *Server) onVariablesRequest(request *dap.VariablesRequest) {
 				kvvar := dap.Variable{
 					Name:         key,
 					EvaluateName: valexpr,
+					Type:         valType, // The type of a key/value pair is the value type.
```

Comment:
I  think thats a great idea!

I had put just the value type because it is what I had seen from other debug adapters (python and typescript), but I think your idea would be more clear. 

I'll add a screenshot for this.